include/yaclib/container/intrusive_node.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+namespace yaclib::container::detail {
+
+template <typename T>
+class IntrusiveNode {
+ public:
+  // Links this node before next in list
+  void LinkBefore(IntrusiveNode* next) noexcept {
+    _prev = next->_prev;
+    _prev->_next = this;
+    _next = next;
+    _next->_prev = this;
+  }
+
+  // Is this node linked in a circular list?
+  bool IsLinked() const noexcept {
+    return _next != nullptr;
+  }
+
+  // Unlink this node from current list
+  void Unlink() noexcept {
```

Comment:
- [ ] Remove runtime conditions

I propose linking the node to itself. Then ifs aren't needed
https://godbolt.org/z/6n938195G